modules/ppcp-vaulting/src/PaymentTokensMigration.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * The payment tokens migration handler.
+ *
+ * @package WooCommerce\PayPalCommerce\Vaulting
+ */
+
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\Vaulting;
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\CreditCardGateway;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\PayPalGateway;
+
+/**
+ * Class PaymentTokensMigration
+ */
+class PaymentTokensMigration {
+	/**
+	 * The payment token factory.
+	 *
+	 * @var PaymentTokenFactory
+	 */
+	private $payment_token_factory;
+
+	/**
+	 * The payment token repository.
+	 *
+	 * @var PaymentTokenRepository
+	 */
+	private $payment_token_repository;
+
+	/**
+	 * The logger.
+	 *
+	 * @var LoggerInterface
+	 */
+	private $logger;
+
+	/**
+	 * PaymentTokensMigration constructor.
+	 *
+	 * @param PaymentTokenFactory    $payment_token_factory The payment token factory.
+	 * @param PaymentTokenRepository $payment_token_repository The payment token repository.
+	 * @param LoggerInterface        $logger The logger.
+	 */
+	public function __construct(
+		PaymentTokenFactory $payment_token_factory,
+		PaymentTokenRepository $payment_token_repository,
+		LoggerInterface $logger
+	) {
+		$this->payment_token_factory    = $payment_token_factory;
+		$this->payment_token_repository = $payment_token_repository;
+		$this->logger                   = $logger;
+	}
+
+	/**
+	 * Migrates user existing vaulted tokens into WC payment tokens API.
+	 *
+	 * @param int $id WooCommerce customer id.
+	 */
+	public function migrate_payment_tokens_for_user( int $id ):void {
+		$tokens          = (array) get_user_meta( $id, 'ppcp-vault-token', true );
```

Comment:
Because of the previous issue #1104, I don't know that we can trust that the tokens stored in `ppcp-vault-token` belong to the main site.